e2e/test/scenarios/embedding/embedding-dashboard.cy.spec.js

```diff
@@ -222,6 +223,32 @@ describe("scenarios > embedding > dashboard parameters", () => {
       cy.contains(".ScalarValue", "2");
     });
   });
+
+  it("should render error message when `params` is not an object (metabase#14474)", () => {
+    cy.get("@dashboardId").then(dashboardId => {
+      cy.request("PUT", `/api/dashboard/${dashboardId}`, {
+        embedding_params: {
+          id: "enabled",
+          name: "enabled",
+          source: "enabled",
+          user_id: "enabled",
+        },
+        enable_embedding: true,
+      });
+
+      const invalidParamsValue = [];
+      const payload = {
+        resource: { dashboard: dashboardId },
+        params: invalidParamsValue,
```

Comment:
If we copy the code from Metabase's code panel, this won't be wrong
![image](https://github.com/metabase/metabase/assets/1937582/a6e436af-231e-4465-a449-da8222a808aa)

but since users can just change this to anything, or implement their own for languages that we don't have an example for it's possible to pass the wrong value type.